trace_decoder/tests/trace_decoder_tests.rs

```diff
@@ -0,0 +1,272 @@
+//! These tests rely on the jerigon and cdk erigon witness files as input
+//! To add a new block or range of blocks as test input, do the following:
+//! 1. Run rpc tool to fetch block witness:
+//!```
+//!     cargo run --bin rpc fetch --rpc-url <node_rpc_endpoint> --start-block <start> --end-block <end> ./b<number>_<network>.json
+//! ```
+//! 2. Download the header file for the block or range of blocks:
+//!```
+//!     file_name = b<number>_<network>_header.json
+//!     echo "[" > $file_name && cast rpc eth_getBlockByNumber "0x<block_number>" 'false' --rpc-url <node_rpc_endpoint>  >> $file_name && echo "]" >> $file_name
+//! ```
+
+use std::time::Duration;
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+use alloy_rpc_types_eth::Header;
+use anyhow::Context as _;
+use evm_arithmetization::prover::testing::simulate_execution;
+use evm_arithmetization::GenerationInputs;
+use itertools::Itertools;
+use log::{info, warn};
+use mpt_trie::partial_trie::PartialTrie;
+use plonky2::field::goldilocks_field::GoldilocksField;
+use plonky2::util::timing::TimingTree;
+use plonky2_maybe_rayon::*;
+use pretty_env_logger::env_logger::{try_init_from_env, Env, DEFAULT_FILTER_ENV};
+use prover::BlockProverInput;
+use rstest::rstest;
+use trace_decoder::OtherBlockData;
+
+type F = GoldilocksField;
+
+const JERIGON_WITNESS_DIR: &str = "tests/data/witnesses/zero_jerigon";
+//const CDK_ERIGON_WITNESS_DIR: &str =
+// "tests/data/witnesses/hermez_cdk_erigon";
```

Comment:
these are type2 data? Maybe add a TODO specifying when to get back at them.
Could be addressed on `feat/type2` PR's base branch one rebased with this work.